shopwild-frontend/src/components/AllProducts/Product/Product.jsx

```diff
@@ -0,0 +1,169 @@
+import "./Product.css";
+
+import { useNavigate, Link } from "react-router-dom";
+
+import { useData } from "../../../context/data-context";
+
+import serverRequests from "../../../server/serverRequests";
+
+const Product = (props) => {
+  const { dispatch } = useData();
+  const navigate = useNavigate();
+
+  const utilToast = (setToast, setToastMessage, message) => {
+    setToast(true);
+    setToastMessage(message);
+    setTimeout(() => {
+      setToast(false);
+    }, 2000);
+  };
+
+  const clickHandler = async (e) => {
+    if (e.target.textContent === "ADD TO CART") {
+      const { error } = await serverRequests({
+        requestType: "post",
+        url: `${process.env.REACT_APP_BACKEND}/cart/607d92eee69f8b99745ef728`,
+        data: { products: [{ product: props.id, quantity: 1 }] },
+      });
+
+      if (!error) {
+        dispatch({
+          type: "ADD_TO_CART",
+          payload: { product: { ...props }, quantity: 1 },
+        });
+        utilToast(props.setToast, props.setToastMessage, "Added to Cart!");
+      }
+    } else if (e.target.textContent === "MOVE TO CART") {
+      const { error } = await serverRequests({
+        requestType: "post",
+        url: `${process.env.REACT_APP_BACKEND}/cart/607d92eee69f8b99745ef728`,
+        data: { products: [{ product: props.id, quantity: 1 }] },
+      });
+
+      if (!error) {
+        dispatch({
+          type: "ADD_TO_CART",
+          payload: { product: { ...props }, quantity: 1 },
+        });
+
+        const { error } = await serverRequests({
+          requestType: "post",
+          url: `${process.env.REACT_APP_BACKEND}/wishlist/607d92eee69f8b99745ef728`,
+          data: { products: [{ product: props.id, isWishlisted: false }] },
+        });
+
+        if (!error) {
+          dispatch({ type: "REMOVE_FROM_WISHLIST", payload: props.id });
+          utilToast(props.setToast, props.setToastMessage, "Moved to Cart!");
+        }
+      }
+    } else if (e.target.textContent === "GO TO CART") {
+      navigate("/cart");
+    }
+  };
+
+  const removeFromWishlistHandler = async () => {
+    const { error } = await serverRequests({
+      requestType: "post",
+      url: `${process.env.REACT_APP_BACKEND}/wishlist/607d92eee69f8b99745ef728`,
+      data: { products: [{ product: props.id, isWishlisted: false }] },
+    });
+
+    if (!error) {
+      dispatch({ type: "REMOVE_FROM_WISHLIST", payload: props.id });
+      utilToast(
+        props.setToast,
+        props.setToastMessage,
+        "Removed from wishlist!"
+      );
+    }
+  };
```

Comment:
As of now you are displayong the toast once the api call is done when you get a response from the server
As soon as a user clicks on ADD_TO_CART try showing a toast like "Item is being added to cart"